### PR TITLE
[IMP] Getting rid of _quant_create method override from product_unique_serial module

### PR DESCRIPTION
**Pull request by hbto**
[IMP] Getting rid of _quant_create method override from product_unique_serial module
[IMP] Fixing tests on product_unique_serial module because of the delete override

### CODE REVIEW

**Comment by hbto** [comment]
@nhomar @moylop260 

Method _quant_create just got deleted from this module and
modules test only got a few changes to fit with the new changes.

Could you please review and/or merge this PR

Regards

File: product_unique_serial/model/stock.py [hunk 1]
```diff
@@ -54,78 +54,25 @@ def _get_last_location_id(self):
...
                     ('company_id', '=', line.company_id.id),
                     ('location_id', '=', line.location_id.id)])
                 if line.qty > 1:
                     raise ValidationError(_(
-                        'The product %s has active unique lot and you '
-                        'try set %s in the lot %s.\nRemmember:\n %s' % (
-                            line.product_id.name, line.qty,
-                            line.lot_id.name, note)))
+                        'You should only receive by the piece with the same '
+                        'serial number'))
```
Comment (nhomar) [inline product_unique_serial/model/stock.py]: As you see hbto you are not showing proper information of the product failling here.

Can you concatenate properly?

File: product_unique_serial/model/stock.py [hunk 2]
```diff
@@ -54,78 +54,25 @@ def _get_last_location_id(self):
...
                 elif sum([x.qty for x in quants]) > 1:
                     raise ValidationError(_(
-                        'The product %s has active unique lot and you '
-                        'try set more products in the same lot '
-                        '%s.\nRemmember:\n %s' % (
-                            line.product_id.name, line.lot_id.name, note)))
+                        'The serial number %s can only belong to a single '
+                        'product in stock') % line.lot_id.name)
```
Comment (nhomar) [inline product_unique_serial/model/stock.py]: We need the product involved also.

**Comment by nhomar** [comment]
I like the new approach.

What We need is send the warning with all the information related (product and serial) involved in the warning.

**Comment by hbto** [comment]
> As you see hbto you are not showing proper information of the product failling here.
> Can you concatenate properly?

I am using the same approach used earlier by avoiding to break the least what is already working.

If some improve is needed I want to do that in a second round,

I am trying to reach an stable process, by changing the test & validation at all I will be causing
too much noise upwards in other modules and integrations in the customer side.

So, yes you are right, Exceptions and Test needs better wording, but that can be done
in a second round.

Regards

File: product_unique_serial/model/stock.py (hunk 1 above)
Comment (hbto) [inline product_unique_serial/model/stock.py]: I am using the same lines that were used before,

https://github.com/Vauxoo/stock-logistics-workflow/pull/14/files#diff-740d05815bfc94b847d7e0d9d485423fL94

In order to be too much hassle the module's test itself and other test upwards

**Comment by hbto** [comment]
@nhomar Could you please review and/or merge

File: product_unique_serial/tests/test_for_unicity.py [hunk 3]
```diff
@@ -30,6 +30,20 @@
...
+    u'is fixed against the quantity in the serial number (lot) and not '
+    u'against the quantity in full of the product.')
+
+msg_greater = _(
+    u'Product %s has been configured to use unique lots. '
+    u'You are trying to set %s items in lot %s. %s')
+
+msg_increase = _(
+    u'Product %s has been configured to use unique lots. '
+    u'You are trying to increase %s items in the lot %s. %s')
```
Comment (nhomar) [inline product_unique_serial/tests/test_for_unicity.py]: Avoid global variables.

Can you please add them as attributes in the class and they will be available in all the class, use this kind of variables is a bad practice.

**Comment by nhomar** [comment]
For me is ok all (thanks) I just have a little comment in the global variables used for the long texts.

The correct approach is add them as new attributes to the class.

``` python
class name(models.Model):
    _your_var = _(----....long text)
    fields_name = fields.Text(help=self._your_var)
# in the method
    def method(self):
        ....
        self._your_var
```

File: product_unique_serial/tests/test_for_unicity.py (hunk 3 above)
Comment (hbto) [inline product_unique_serial/tests/test_for_unicity.py]: Thanks for your advice.

Doing

**Comment by hbto** [comment]
@nhomar 
Your latest advice has been taken into account.

Please recheck.

Regards

File: product_unique_serial/tests/test_for_unicity.py (hunk 3 above)
Comment (hbto) [inline product_unique_serial/tests/test_for_unicity.py]: BTW 

Do pylint or any other of our tools check for this bad practice,
or Have I overlooked the check?

@moylop260

**Comment by nhomar** [comment]
As a last step can you make a dummy PR to yoytec please.!

**Comment by hbto** [comment]
@nhomar 
That is already running since first day at:
https://github.com/Vauxoo/yoytec/pull/1307

Regards

**Comment by hbto** [comment]
@nhomar 

Please review runbot for yoytec on this PR.
http://runbot.vauxoo.com/runbot/static/build/12970-1307-eadbee/logs/job_20_test_all.txt

being this the regarding pull request
https://github.com/Vauxoo/yoytec/pull/1307

this has already gone green

File: product_unique_serial/model/stock.py [hunk 4]
```diff
@@ -54,78 +54,34 @@ def _get_last_location_id(self):
...
-                raise exceptions.Warning(_('The serial number %s can only '
-                                           'belong to a single product in '
-                                           'stock') % lot_name)
-        return super(StockQuant, self)._quant_create(
-            qty, move, lot_id, owner_id, src_package_id,
-            dest_package_id, force_location_from, force_location_to)
-
     @api.multi
     @api.constrains('product_id', 'lot_id', 'qty')
     def _check_inicity_lot_product(self):
```
Comment (nhomar) [inline product_unique_serial/model/stock.py:59]: s/inicity/uniqueness/g

s/_check_inicity_lot_product/_check_uniqueness_lot/g

**Comment by nhomar** [comment]
It is Ok for me, just a little typo to merge, can you @hugho-ad check this please to merge ASAP

**Comment by hugho-ad** [comment]
I'm not full in context for a reviewing, maybe you are askying for Moy's reviewing

Regards

**Comment by nhomar** [comment]
Sorry I meant @hbto

File: product_unique_serial/model/stock.py (hunk 4 above)
Comment (hbto) [inline product_unique_serial/model/stock.py:59]: > s/inicity/uniqueness/g
> 
> s/_check_inicity_lot_product/_check_uniqueness_lot/g

@nhomar 
A long time ago that was taken care of

check https://github.com/Vauxoo/stock-logistics-workflow/pull/13

Regards

**Pull request merged**